src/components/BookItem.jsx

```diff
@@ -2,23 +2,43 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useDispatch } from 'react-redux';
 import { deleteBook } from '../redux/books/booksSlice';
+import { CircularProgressBar } from './Icons';
 
 const BookItem = ({ book }) => {
+  const randomPercentage = Math.floor(Math.random() * 101);
   const dispatch = useDispatch();
   return (
-    <>
-      <span>
-        <strong>Title:</strong>
-        {book.title}
-        ,
-        <strong>Author:</strong>
-        {book.author}
-        ,
-        <strong>Category:</strong>
-        {book.category}
-      </span>
-      <button type="button" onClick={() => dispatch(deleteBook(book))}>Delete</button>
-    </>
+    <div className="single-book-container">
+      <div className="single-book">
+        <div className="book-info">
+          <span className="book-category">{book.category}</span>
+          <span className="book-title">{book.title}</span>
+          <span className="book-author">{book.author}</span>
+        </div>
+        <div className="btn-container">
+          <button type="button">Comments</button>
+          <span>|</span>
+          <button type="button" onClick={() => dispatch(deleteBook(book))}>Remove</button>
+          <span>|</span>
+          <button type="button">Edit</button>
+        </div>
+      </div>
+      <div className="mid-section">
+        <CircularProgressBar percentage={randomPercentage} />
+        <div>
+          <span className="percentage-text">
+            {randomPercentage}
+            %
+          </span>
+          <p className="completed-text">Completed</p>
+        </div>
+      </div>
+      <div className="end-section">
+        <span className="current-chapter">Current Chapter</span>
+        <span className="chapter-number">Chapter 1</span>
+        <button type="button" className="update-btn">Update Progress</button>
+      </div>
```

Comment:
- [ ] Absolutely! You've made significant progress on the layout. However, I'd like to point out that according to the design specifications, there should be a horizontal line separating the `mid-section` and the `end-section`. To ensure your project aligns perfectly with the design, I kindly recommend adding that horizontal line exactly as depicted in the design. This subtle addition will contribute to the overall visual coherence of your layout.
![image](https://github.com/AliRezaBmeDu/bookstore/assets/55840999/b176325d-d5b3-463a-a8fc-0723e312cac2)

Great work so far, and thank you for your dedication to bringing the design to life! 🌟📏